sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/Item.kt

```diff
@@ -1,29 +1,48 @@
 package co.nimblehq.sample.compose.ui.screens.home
 
-import androidx.compose.foundation.clickable
+import androidx.compose.foundation.*
 import androidx.compose.foundation.layout.*
-import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.material.*
+import androidx.compose.runtime.*
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import co.nimblehq.sample.compose.R
 import co.nimblehq.sample.compose.model.UiModel
+import co.nimblehq.sample.compose.ui.theme.*
 import co.nimblehq.sample.compose.ui.theme.AppTheme.dimensions
-import co.nimblehq.sample.compose.ui.theme.ComposeTheme
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun Item(
     uiModel: UiModel,
-    onClick: (UiModel) -> Unit
+    onClick: (UiModel) -> Unit,
+    onLongClick: (UiModel) -> Unit
 ) {
-    Row(
+    var expanded by remember { mutableStateOf(false) }
+
+    Box(
         modifier = Modifier
             .fillMaxWidth()
-            .clickable(onClick = { onClick(uiModel) })
+            .combinedClickable(
+                onClick = { onClick(uiModel) },
+                onLongClick = { expanded = !expanded }
+            )
     ) {
         Text(
-            modifier = Modifier.padding(dimensions.spacingNormal),
+            modifier = Modifier
```

Comment:
@hoangnguyen92dn @kaungkhantsoe @Tuubz May I get your opinion on this 🙏 

What do you guys think if we provide `modifier` from the root/parent(**HomeScreen**) of `Composable` that we want to use this `Composable`(**Item**)? 🤔 

If I'm not wrong, it might affect when `re-composition`  in the case when we have another screen that also uses the `Item` function or when we have a child function that's separate from the parent.

```suggestion
            modifier = modifier
```

Reply:
@Wadeewee Are you suggesting we should pass `modifier` through the constructor? 💭 

Sorry for asking for clarification, I'm trying to better understand your suggestion 🙇 

```
fun Item(
    uiModel: UiModel,
    onClick: (UiModel) -> Unit,
    onLongClick: (UiModel) -> Unit,
    modifier: Modifier // this is new
)
```

Reply:
I am not sure I understand you @Wadeewee. Can you elaborate more on why `modifier` would affect recomposition?

Reply:
> Are you suggesting we should pass modifier through the constructor?

Yes, @Tuubz, and Sorry 🙏 If I write the message not clearly.

Reply:
@Wadeewee If we do that, then it means all the components in this screen will make use of the same `modifier` 

Sounds risky and difficult to manage 💭

Reply:
We can pass the modifier easily but I think it depends on how you will use it. I don't think `modifier` can affect `recomposition` in this case. Currently, it is just a simple UI sample so I suggest we should leave it as it is. 🙏 cc @Wadeewee @Tuubz

Reply:
@kaungkhantsoe @Tuubz, I might be wrong that I said it could effect `re-composition` 🙇‍♀️  but here're my ref why we should have a `modifier` as a parameter when we create a `Composable` function(**Item**), and basically it's like another base `Composable`, e.g., (Text, Colum, Button and so on) all of them have a parameter `modifier`

> * @param modifier The modifier to be applied to the layout. // ---(Layout = HomeScreen)--- //
 
- https://stackoverflow.com/a/71465727
- https://stackoverflow.com/a/71475970
- [Jetpack-Compose-Playground](https://foso.github.io/Jetpack-Compose-Playground/general/modifier/)

![Screen Shot 2566-02-01 at 13 12 45](https://user-images.githubusercontent.com/28002315/215965678-9b3987b3-4a79-4ed7-9ade-175afd410703.png)
- And I noticed from the [Follow best practices](https://developer.android.com/jetpack/compose/performance/bestpractices) on the Composable `ContactList` they also have a parameter `modifier: Modifier = Modifier` as well.

![Screen Shot 2566-02-01 at 13 20 03](https://user-images.githubusercontent.com/28002315/215966782-fcfa85b5-6c38-471e-9241-0b514cf75102.png)

Actually, I just want to know what you think about this cc @hoangnguyen92dn

Reply:
@Wadeewee I see 

Adding `Modifier` to the constructor is good for flexibility/reusability, which I agree can be good for some cases 👍 

But I'm not 100% convinced this should be the case everywhere 💭 

So it might be better to create an RFC for this, what do you think? 🙏

Reply:
Yeah, I agree @Tuubz, we should have an `RFC` for this, I don't expect to update on this PR too 🚀 just ask for your idea 🙏